haystack/dataclasses/chat_message.py

```diff
@@ -99,3 +99,51 @@ def from_function(cls, content: str, name: str) -> "ChatMessage":
         :returns: A new ChatMessage instance.
         """
         return cls(content, ChatRole.FUNCTION, name)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Converts ChatMessage into a dictionary.
+
+        :returns:
+            Serialized version of the object.
+        """
+        data = asdict(self)
+        data["role"] = self.role.value
+
+        meta = data.pop("meta")
+        return {**data, **meta}
```

Comment:
The metadata could contain a key with the same name as one of the dataclass' field. We should keep it as a nested dict.

Reply:
True, I did not think about it. I believed that as it was the default in documents, we could just keep the flattened standard here.

Reply:
Let's keep it nested.

Reply:
okey